### PR TITLE
feat(pattern/select): create select pattern

### PR DESCRIPTION
**Pull request by marineFabien**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/adeo/design-system--styleguide/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

### CODE REVIEW

File: src/pages/Components/Select/specs.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+| State       | Property      | Token                |
+| ----------- | ------------- | -------------------- |
+| **Default** | padding       | `mu-100` (1rem)      |
+| **Default** | border-radius | `radius.m` (4px)     |
+| **Default** | border-width  | `border.s` (1px)     |
+| **Default** | font-size     | `size.font.05`(16px) |
+
+### CSS Tokens : state colors
+
+> **You must always use the tokens, the colors are only there for that information.**
```
Comment (tiagoux) [inline src/pages/Components/Select/specs.md]: You must always use the tokens, the colors are listed here for information purposes only

File: packages/styles/components/_c.select.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  @include select-background-arrow();
+
+  border: $select-border-width solid $color-input-border;
+  border-radius: $select-border-radius;
+  color: $color-input-text;
+  width: 100%;
+  padding:
+    calc(#{$mu075} - 0.125em - #{$select-border-width}) $mu075
+    calc(#{$mu075} + 0.125em - #{$select-border-width});
+  transition: border-color ease 200ms, border ease 200ms;
```
Comment (gael-boyenval) [inline packages/styles/components/_c.select.scss]: double

File: packages/styles/components/_c.select.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
+.mc-select {
+  @include reset-input;
+  @include set-font-scale('05');
+  @include select-background-arrow();
+
+  border: $select-border-width solid $color-input-border;
+  border-radius: $select-border-radius;
+  color: $color-input-text;
+  width: 100%;
+  padding:
```
Comment (gael-boyenval) [inline packages/styles/components/_c.select.scss]: comment

File: src/pages/Components/Select/code.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+@import 'settings-tools/_all-settings';
+@import 'components/_c.select';
+```
+
+## Basic usage
+
+Apply the classes `mc-select` on the native element of the select as in the following example :
+
+```html
+<div class="example">
```
Comment (gael-boyenval) [inline src/pages/Components/Select/code.md]: not example

File: src/pages/Components/Select/code.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+
+<preview path="src/pages/Components/Select/previews/Select-all-states"></preview>
+
+## Default option (as a placeholder)
+ By default, the first option is displayed. You can use it as a placeholder, but remember to distinguish it from the rest of options and to add an empty value attribute.
+ 
+ ```html
+<select class="mc-select">
+  <option value="">--select one option--</option>
```
Comment (gael-boyenval) [inline src/pages/Components/Select/code.md]: spaces between dashes

File: src/pages/Components/Select/index.md [hunk 6]
```diff
@@ -30,8 +30,9 @@ order: 6
 </hintitem>
```
Comment (gael-boyenval) [inline src/pages/Components/Select/index.md]: add default option text convention : `-- Select an option --`

**Pull request merged**